HHS/Head-Start-TTADP
# [Prod] TR dashboard, resource dashboard updates, validation fixes

## Description of change

### 1185
Change validation behavior when picking activity recipients on activity report

### 2570
Rewrite resource dashboard backend to improve performance

### Others
All the other tickets referenced refer to the first part of the front and back end of the training report dashboard

## How to test

- To test 1185, attempt to pick a recipient from an activity report without first choosing the type. You will see validation appear.
- To test 2570, simply view the resource dashboard and observe the performance gains
- To test the others, view the TR dashboard and validate that the widgets display and function correctly and that the data is correct.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-1185
* https://jira.acf.gov/browse/TTAHUB-2752
* https://jira.acf.gov/browse/TTAHUB-2756
* https://jira.acf.gov/browse/TTAHUB-2758
* https://jira.acf.gov/browse/TTAHUB-2757
* https://jira.acf.gov/browse/TTAHUB-2753
* https://jira.acf.gov/browse/TTAHUB-2755

## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
